File: card-service/src/main/java/a/gleb/cardservice/configuration/properties/CardServiceConfigurationProperties.java
package a.gleb.cardservice.configuration.properties;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.web.cors.CorsConfiguration;

@Getter
@Setter
@ConfigurationProperties("card-service")
public class CardServiceConfigurationProperties {

    private Cors corsConfiguration;

    public static class Cors extends CorsConfiguration {}
}

File: card-service/src/main/kotlin/a/gleb/cardservice/configuration/properties/CardServiceConfigurationProperties.kt
package a.gleb.cardservice.configuration.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.web.cors.CorsConfiguration

@ConfigurationProperties("card-service")
data class CardServiceConfigurationProperties(
    var corsConfiguration: Cors,
    var securityConstraints: List<SecurityConstraints>
)

class Cors: CorsConfiguration()

data class SecurityConstraints(
    var securityCollections: List<SecurityCollection>,
    var authRoles: List<String>
)

data class SecurityCollection(
    var patterns: List<String>,
    var methods: List<String>?
)
